File: CS-Examples/09_Charts/CreateChartBasedOnPivotTable.cs
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet.PivotTables;
using System;
using System.Windows.Forms;

namespace CreateChartBasedOnPivotTable
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            // Load an excel file including pivot table
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\PivotTable.xlsx");
          
            // Get the sheet in which the pivot table is located
            Worksheet sheet = workbook.Worksheets[0];

            // Get the pivot table
            XlsPivotTable pt = sheet.PivotTables[0] as XlsPivotTable;

            // Add a chart based on the pivot table to the second worksheet
            workbook.Worksheets[1].Charts.Add(ExcelChartType.BarClustered, pt);

            // Save the document
            string output = "CreateChartBasedOnPivotTable.xlsx";
            workbook.SaveToFile(output, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the document
            FileViewer(output);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: VB-Examples/09_Charts/CreateChartBasedOnPivotTable.vb
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet.PivotTables
Imports System.ComponentModel
Imports System.Text

Namespace CreateChartBasedOnPivotTable
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file "PivotTable.xlsx" from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\PivotTable.xlsx")

            ' Get the first worksheet from the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Retrieve the first PivotTable from the worksheet and cast it to XlsPivotTable type
            Dim pt As XlsPivotTable = TryCast(sheet.PivotTables(0), XlsPivotTable)

            ' Add a clustered bar chart to the second worksheet, based on the PivotTable data
            workbook.Worksheets(1).Charts.Add(ExcelChartType.BarClustered, pt)

            ' Specify the output file name as "CreateChartBasedOnPivotTable.xlsx"
            Dim output As String = "CreateChartBasedOnPivotTable.xlsx"

            ' Save the modified workbook to the specified file path, using Excel 2013 format
            workbook.SaveToFile(output, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(output)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
